index.bs

```diff
@@ -0,0 +1,225 @@
+        1. Set <var>element</var>'s <a>current visual representation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>current frame starting point</a> and <a>previous frame starting point</a>. 
+        1. Set <var>element</var>'s <var>shiftFraction</var> to:
+            1. 0 if <a>previous frame starting point</a> is null.
+            1. The max distance the element has moved in any direction computed by:
+                1. Let <var>verticalDistance</var> be abs(<a>previous frame starting point</a>'s y-value - <a>current frame starting point</a>'s y-value) / <var>viewportDistance</var>.
+                1. Let <var>horizontalDistance</var> be abs(<a>previous frame starting point</a>'s x-value - <a>current frame starting point</a>'s x-value) / <var>viewportDistance</var>.
+                1. Take the max(<var>verticalDistance</var>, <var>horizontalDistance</var>).
+        1. If <var>unstable</var> is true, add a tuple to <var>unstableElements</var> with:
+            1. <a>current visual representation</a> of <var>element</var>
+            1. <a>previous visual representation</a> of <var>element</var>
+            1. <var>shiftFraction</var> of <var>element</var>:
+        1. Set <var>element</var>'s <a>previous frame starting point</a> point to <a>current frame starting point</a>.
+        1. Set <var>element</var>'s <a>previous visual representation</a> to the <a>current visual representation</a>.
+    1. Call <a>compute the layout shift</a> with <var>unstableElements</var>.
+</div>
+
+Compute the layout shift {#sec-compute-layout-shift}
+-----------------------------------------------------
+
+<div algorithm="compute the layout shift">
+    When asked to to <dfn export>compute the layout shift</dfn>, with <var>unstableElements</var>, a list of impacted <a>element</a>'s on the page, as input, run the following steps:
+
+    1. Let <var>impactedRegion</var> be a two-dimensional geometric region which is initially empty (that is, it contains no points).
+    1. Let <var>maxShiftFraction</var> be initially set to 0.
+    1. For each tuple <var>tuple</var> in <var>unstableElements</var>:
+        1. If <a>previous visual representation</a> is not null, expand <var>impactedRegion</var> to additionally contain all points that lie within <var>tuple</var>'s <a>current visual representation</a>, <var>tuple</var>'s <a>previous visual representation</a>, or both.
+        1. Set <var>maxShiftFraction</var> to <var>shiftFraction</var> if <var>tuple</var>’s <var>shiftFraction</var> > <var>maxShiftFraction</var>.
+    1. Let <var>impactedFraction</var> be the area of the <a>initial containing block</a> of the <var>impactedRegion</var> divided by the area of the current <a>viewport</a>.
```

Comment:
"initial containing block of the impactedRegion" doesn't make sense.  This should just be "area of impactedRegion".

But also I think we are missing a step to clip the region to the viewport, e.g. (above this line):

"Subtract from impactedRegion all points that lie outside of the current viewport."

Reply:
Done.